Objects/unicodeobject.c

```diff
@@ -2272,34 +2269,85 @@ PyUnicode_FromString(const char *u)
     return PyUnicode_DecodeUTF8Stateful(u, (Py_ssize_t)size, NULL, NULL);
 }
 
+
 PyObject *
 _PyUnicode_FromId(_Py_Identifier *id)
 {
-    if (!id->object) {
-        id->object = PyUnicode_DecodeUTF8Stateful(id->string,
-                                                  strlen(id->string),
-                                                  NULL, NULL);
-        if (!id->object)
+    PyInterpreterState *interp = _PyInterpreterState_GET();
+    struct _Py_unicode_ids *ids = &interp->unicode.ids;
+
+    // Copy the index since _Py_Identifier.index is declared as volatile
+    Py_ssize_t index = id->index;
+    if (index < 0) {
```

Comment:
Also, I suggest to use 0 as an initial value.

Reply:
See my previous comment: https://github.com/python/cpython/pull/20058#discussion_r424600281

Reply:
_Py_atomic_int_get() is now used.